src/python/grpcio/grpc/beta/implementations.py

```diff
@@ -105,10 +106,37 @@ def metadata_call_credentials(metadata_plugin, name=None):
     A CallCredentials object for use in a GRPCCallOptions object.
   """
   if name is None:
-    name = metadata_plugin.__name__
```

Comment:
I've updated the behavior to check for the function name first, and then fall back on the class name.